common/repo/GitRepo.java

```diff
@@ -366,33 +299,65 @@ public void removeBranch(String branch) {
             .collect(Collectors.toList()));
   }
 
+  @Override
   public String getFileContents(String commitId, String path) {
-    try (ObjectReader objectReader = jGitRepo.newObjectReader();
-        RevWalk revWalk = new RevWalk(objectReader)) {
-      final ObjectId objectId = jGitRepo.resolve(commitId);
-      RevCommit revCommit = revWalk.parseCommit(objectId);
-      RevTree revTree = revCommit.getTree();
-      byte[] data;
-      try (TreeWalk treeWalk = TreeWalk.forPath(objectReader, path, revTree)) {
-        if (treeWalk == null) {
-          throw new IllegalStateException(
-              String.format("TreeWalk is null for commitId %s and path %s", commitId, path));
-        }
-        // Index 0 should have file data
-        data = objectReader.open(treeWalk.getObjectId(0)).getBytes();
-      }
-      return new String(data, UTF_8);
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    CommandResult commandResult = runCommand("show " + commitId + ":" + path);
+    String result = commandResult.stdout;
+    int lastIndexOfNewLineSymbol = result.lastIndexOf("\n");
+    if (lastIndexOfNewLineSymbol >= 0) {
+      result =
+          new StringBuilder(result)
+              .replace(lastIndexOfNewLineSymbol, lastIndexOfNewLineSymbol + 1, "")
+              .toString();
     }
+    return result;
+  }
+
+  @Override
+  public String currentBranch() {
+    return runCommand("rev-parse --abbrev-ref HEAD").stdout.trim();
   }
 
   public void init() {
     runCommand("init");
   }
 
-  public String currentBranch() {
-    return runCommand("rev-parse --abbrev-ref HEAD").stdout.trim();
+  public void addFile(String path) {
+    runCommand("add " + path);
+  }
+
+  public String getHeadCommitId() {
+    CommandResult commandResult = runCommand("rev-parse HEAD");
+    return commandResult.stdout.replace("\n", "");
+  }
+
+  private void switchToMasterBranch() {
+    if (!currentBranch().equals("master")) {
+      System.out.println("You are not in the branch master");
```

Comment:
No need to print these. It's normal not to be in the master branch sometimes, and the calling code can decide to log.

Reply:
Done.